test/client/fixtures/sandbox/node/dom-processor-test.js

```diff
@@ -31,42 +32,53 @@ test('iframe', function () {
 });
 
 test('link in iframe', function () {
-    var $iframe    = $('<iframe id="test1">').appendTo('body');
-    var iframeBody = $iframe[0].contentDocument.body;
-    var $link      = $('<a href="/index.html">').appendTo('body');
+    var iframe = nativeMethods.createElement.call(document, 'iframe');
+
+    iframe.id = 'test';
+    nativeMethods.appendChild.call(document.body, iframe);
+
+    var link = nativeMethods.createElement.call(document, 'a');
 
-    // HACK: IE.
+    link.href = '/index.html';
+    nativeMethods.appendChild.call(document.body, link);
+
+    var iframeBody = iframe.contentDocument.body;
+
+    // NOTE: In IE9, iframe's contentDocument does not have a 'body' element.
+    // So, we need to create it manually.
     if (!iframeBody) {
-        $iframe[0].contentDocument.write('<body></body>');
-        iframeBody = $iframe[0].contentDocument.body;
+        iframeBody = nativeMethods.createElement.call(iframe.contentDocument, 'body');
+        nativeMethods.appendChild.call(iframe.contentDocument, iframeBody);
     }
 
+
     iframeBody.innerHTML = '<a href="/index.html"></a>';
 
     domProcessor.processElement(iframeBody.childNodes[0], urlUtils.convertToProxyUrl);
-    domProcessor.processElement($link[0], urlUtils.convertToProxyUrl);
+    domProcessor.processElement(link, urlUtils.convertToProxyUrl);
 
     strictEqual(urlUtils.parseProxyUrl(iframeBody.childNodes[0].href).resourceType, 'i');
-    ok(!urlUtils.parseProxyUrl($link[0].href).resourceType);
+    ok(!urlUtils.parseProxyUrl(link.href).resourceType);
 
-    $iframe.remove();
-    $link.remove();
+    iframe.parentNode.removeChild(iframe);
+    link.parentNode.removeChild(link);
```

Comment:
We should use  native `createElement` and `appendChild` to prevent overriding elements.
On `removeChild` we don't override elements.